SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp

```diff
@@ -7,8 +7,9 @@
 BodyPartByParticleTriMesh::BodyPartByParticleTriMesh(SPHBody* body, string body_part_name, TriangleMeshShape* triangle_mesh_shape)
 : BodyPartByParticle(body, body_part_name)
 {	
-	body_part_shape_ = new ComplexShape(body_part_name);
-	body_part_shape_->addTriangleMeshShape(triangle_mesh_shape, ShapeBooleanOps::add);
+	ComplexShapeTriangleMesh *mesh = new ComplexShapeTriangleMesh();
```

Comment:
where this ptr is cleaned? please use unique ptr

Reply:
ComplexShapeTriangleMesh pointer is constructed by using unique_ptr